client/src/components/course-results-view/SisuExportDialog.tsx

```diff
@@ -94,66 +80,53 @@ export default function SisuExportDialog(props: {
-  const [assessmentDate, setAssessmentDate]: State<string | null> = useState<string | null>(null);
-  const [completionLanguage, setCompletionLanguage]: State<string | null> =
-    useState<string | null>(null);
-
-  // useEffect in charge of handling the back-to-back alerts
-  // makes the previous disappear before showing the new one
-  useEffect(() => {
-    if (snackPack.length && !messageInfo) {
-      setMessageInfo({ ...snackPack[0] });
-      setSnackPack((prev: Array<Message>) => prev.slice(1));
-      setAlertOpen(true);
-    } else if (snackPack.length && messageInfo && alertOpen) {
-      setAlertOpen(false);
-    }
-  }, [snackPack, messageInfo, alertOpen]);
-
-  async function exportSisuCsvGrades(): Promise<void> {
-    setSnackPack((prev: Array<Message>) => [...prev, loadingMsg]);
-
-    try {
-      if (courseId && assessmentModelId) {
-        const params: {
-          completionLanguage?: string,
-          assessmentDate?: string,
-          studentNumbers: Array<string>
-        } = {
-          completionLanguage: completionLanguage ?? undefined,
-          assessmentDate: assessmentDate ?? undefined,
-          studentNumbers: props.selectedStudents.map((student: FinalGrade) => student.studentNumber)
-        };
-
-        const data: BlobPart = await exportSisuCsv(
-          courseId, assessmentModelId, params
-        );
-
-        // Create a blob object from the response data
-        const blob: Blob = new Blob([data], { type: 'text/csv' });
-
-        const link: HTMLAnchorElement = document.createElement('a');
-        link.href = URL.createObjectURL(blob);
-        // Set file name.
-        link.download = `grades_course_${courseId}_assessment_model_${assessmentModelId}.csv`;
-        // Download file automatically to the user's computer.
-        link.click();
-        URL.revokeObjectURL(link.href);
-        link.remove();
+  const [assessmentDate, setAssessmentDate]: State<string | undefined> =
+    useState<string | undefined>(undefined);
+  const [completionLanguage, setCompletionLanguage]: State<string | undefined> =
+    useState<string | undefined>(undefined);
+
+  const exportSisuGradeCsv: UseQueryResult<BlobPart> = useExportSisuGradeCsv(
+    courseId, assessmentModelId,
+    {
+      completionLanguage: completionLanguage,
+      assessmentDate: assessmentDate,
+      studentNumbers: props.selectedStudents.map((student: FinalGrade) => student.studentNumber)
+    },
+    { enabled: false }
+  );
 
-        setSnackPack((prev: Array<Message>) => [...prev, successMsg]);
-      }
-    } catch (error: unknown) {
-      console.log(error);
-      setSnackPack((prev: Array<Message>) => [...prev, errorMsg]);
-    } finally {
-      setAlertOpen(false);
+  async function handleExportSisuGradeCsv(): Promise<void> {
+    snackPack.push({
+      msg: 'Fetching Sisu CSV...',
+      severity: 'info'
+    });
+
+    exportSisuGradeCsv.refetch();
```

Comment:
invalidating cache. Exporting to Sisu compatible CSV, if used twice, returns the first exports cached file when exporting the second time. This creates problems if in between export calls the list of student numbers, assessmentdate or completionLanguage is changed. We should some how invalidate the cache if any of these values have been modified between exports.

But because this PR is already quite big, the caching strategies should be planned/implemented on some other PR.

Reply:
New issue: #382